dashboard/app/controllers/activities_controller.rb

```diff
@@ -213,6 +215,23 @@ def track_progress_for_user
     end
   end
 
+  # Record the time spent on a level by creating or updating the UserLevelInfo table
+  # for the UserLevel for the current user, level, and script
+  def record_time_spent(user_level)
+    if user_level
+      user_level_info = UserLevelInfo.find_by(
+        user_level_id: user_level.id
+      )
+    end
+
+    if user_level_info
+      # Add past time spent to current time spent on level
+      user_level_info.update(time_spent: [user_level_info.time_spent + [params[:timeSinceLastMilestone].to_i, 0].max, MAX_INT_MILESTONE].min)
+    else
```

Comment:
What's `MAX_INT_MILESTONE`? Something that already existed?

Reply:
It already existed to make sure we didn't exceed the biggest number we could put in for time so I kept that same practice with timeSinceLastMilestone